File: solutions/go/nth-prime/1/nth_prime.go
package prime

func Nth(n int) (int, bool) {
	if n < 1 {
		return 0, false
	}
	count := 0
	for i := 1; ; i++ {
		if isPrime(i) {
			count++
			if count == n {
				return i, true
			}
		}
	}
}

func isPrime(n int) bool {
	if n < 2 {
		return false
	}
	if n == 2 {
		return true
	}
	if n%2 == 0 {
		return false
	}
	for i := 3; i*i <= n; i += 2 {
		if n%i == 0 {
			return false
		}
	}
	return true
}

File: solutions/clojure/nth-prime/1/src/nth_prime.clj
(ns nth-prime)

(defn- prime?
  [n]
  (if (= n 2)
    true
    (let [max-n #(inc (Math/sqrt n))]
      (empty? (filter #(zero? (mod n %)) (range 2 (max-n)))))))

(defn nth-prime
  [n]
  (if (> n 0)
    (last (take n (filter prime? (iterate inc 2))))
    (throw (IllegalArgumentException. "no 0th prime"))))